### PR TITLE
feat: onboard fullstory cloud mode

### PR DESCRIPTION
**Pull request by yashasvibajpai**
## Description of the change

> Resolves INT-133
Onboard fullstory cloud mode support

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by koladilip — APPROVED** [review]

File: src/cdk/v2/destinations/fullstory/procWorkflow.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        "model": .message.context.device.model,
+      };
+      $.context.payload.context.location = {
+        "ip_address": .message.context.ip,
+        "latitude": .message.properties.latitude,
+        "longitude": .message.properties.longitude,
+        "city": .message.properties.city,
+        "region": .message.properties.region,
+        "country": .message.properties.country,
+      };
```
Comment (sanpj2292) [inline src/cdk/v2/destinations/fullstory/procWorkflow.yaml:69]: Should payload contain one of `browser`, `mobile`, `device` ? Or all 3 at all times ?!

File: src/cdk/v2/destinations/fullstory/procWorkflow.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      $.context.endpoint = "https://api.fullstory.com/v2/users";
+      $.context.payload.properties = .message.traits ?? .message.context.traits;
+      $.context.payload.uid = .message.userId;
+      $.context.payload.email = .message.context.traits.email;
+      $.context.payload.display_name = .message.context.traits.name;
+  
+  - name: trackPayload
+    condition: $.context.messageType == "track"
+    template: |
+      $.context.endpoint = "https://api.fullstory.com/v2/events";
```
Comment (sanpj2292) [inline src/cdk/v2/destinations/fullstory/procWorkflow.yaml:35]: [Q] Does fullstory support different endpoints for EU or US instances ? If we hadn't verified it, can we quickly verify it ?

File: src/cdk/v2/destinations/fullstory/procWorkflow.yaml (hunk 1 above)
Comment (yashasvibajpai) [reply src/cdk/v2/destinations/fullstory/procWorkflow.yaml:69]: No restrictions on `browser`, `mobile` or `device`. They can exist or coexist independently. I have handled cleanup in cleanPayload step ahead.

File: src/cdk/v2/destinations/fullstory/procWorkflow.yaml (hunk 2 above)
Comment (yashasvibajpai) [reply src/cdk/v2/destinations/fullstory/procWorkflow.yaml:35]: Yes, but it is handled from fullstory's end.
As per their [data residency docs](https://developer.fullstory.com/server/getting-started/#data-residency),

> All API requests use the api.fullstory.com domain. They are automatically routed to the correct data center based on what is configured for your FullStory account. You can determine which data center your account is associated with in one of two ways:
> - The FullStory URL you use to login
> - API key format: <data center>.<token>

**Review by sanpj2292 — APPROVED** [review]

**Pull request merged**